main.tf

```
@@ -378,6 +378,15 @@ data "aws_iam_policy_document" "ecs_tasks" {
       type        = "Service"
       identifiers = compact(distinct(concat(["ecs-tasks.amazonaws.com"], var.trusted_principals)))
     }
+
+    dynamic "principals" {
+      for_each = var.trusted_entities
+
+      content {
+        type        = "AWS"
+        identifiers = compact(distinct(var.trusted_entities))
```

You can also remove wrapping compact(distinct(...)) because there will be something provided.

```suggestion
        identifiers = var.trusted_entities
```